src/Microsoft.AspNet.SignalR.JS/jquery.signalR.transports.common.js

```diff
@@ -447,6 +447,9 @@
             async = typeof async === "undefined" ? true : async;
 
             var url = getAjaxUrl(connection, "/abort");
+            
+            if (navigator.sendBeacon) //use the new sendBeacon API cause new Chrome disabled XHR in "onbeforeunload"
```

Comment:
We put braces around all `if`'s regardless of size.

Also, the other code passes in an Authorization header, how do we do this in `sendBeacon`? I don't think it's possible to send headers, which is going to be blocking.

Another option could be to use [`fetch` with `keepAlive`](https://fetch.spec.whatwg.org/#ref-for-dom-request-keepalive) which would allow headers to work.

Reply:
It **is** possible to send header with `sendbeacon`, `fetch` is unreliable in an `unload` that exactly why `sendbeacon` was invented - to send stuff when the windo is about to unload. I will make appropriate changes, hopefully it will take you less than 4 months to review the changes again

Reply:
> I will add `fetch`, but its still unreliable in an `unload` that's exactly why `sendBeacon` was invented - to send stuff when the window is about to unload.

`fetch` **with the `keepAlive`** flag is designed to behave the same way as `sendBeacon`. In fact, Chrome implements `sendBeacon` using `fetch` with `keepAlive` now. So I think this is the appropriate change.

> hopefully it will take you less than 4 months to review the changes again ;)

Yep, apologies for the delay. We're the same team that works on ASP.NET Core and we were busy with the .NET Core 3.0 release. Feel free to ping us if we fall behind here!